aws/aws-how-to/instances/upgrade-to-ubuntu-pro-at-scale-using-tokens-with-ssm.rst

```diff
@@ -0,0 +1,120 @@
+- Machines with SSM agent installed (available by default on Ubuntu 16.04 onwards). If it is properly configured, the servers will appear as Managed Nodes on AWS SSM Fleet Manager (see the reference links for more information)
+- Basic understanding about writing and running SSM Documents (see the reference links for more information)
+
+
+Get a token
+------------
+
+To get an Ubuntu Pro token, you need to create an account in `Ubuntu Pro | Ubuntu <https://ubuntu.com/pro>`_ to see your subscriptions and get your tokens. Anyone can use Ubuntu Pro for free on up to 5 machines, even for production workloads.
+
+Log-in by clicking on *Your subscriptions*. You will be redirected to the following screen where the token is exposed as shown in the orange ellipse:
+
+.. image:: upgrade-to-ubuntu-pro-at-scale-using-tokens-with-ssm-images/0_token_screen.png
+   :align: center
+
+
+Create a SSM Document to attach the token
+------------------------------------------
+
+The SSM Document is just a JSON or YAML file containing the script in steps or sections.
+
+    Note:
+    As the token is personal, it is not recommended to hard-code it into the code but add it as a user parameter.
+
+This is the full code:
+
+.. code-block::
+
+    ---
+    schemaVersion: '2.2'
+    description: "Enable Pro services using a Pro token"
+    parameters:
+    token:
+        type: String
+    mainSteps:
+    - action: aws:runShellScript
+    name: update
+    inputs:
+        runCommand:
+        - "sudo apt-get update && sudo apt-get upgrade -y"
+    - action: aws:runShellScript
+    name: attachToken
+    inputs:
+        runCommand:
+        - "sudo pro attach {{ token }}"
+    - action: aws:runShellScript
+    name: updateWithESM
+    inputs:
+        runCommand:
+        - "sudo apt-get update && sudo apt-get upgrade -y"
+
+
+The token will be requested to the user when running this document, then it will execute the rest of the code:
+
+    1. Run an update (to make sure the pro agent is at the latest version)
+    2. Attach the token
+    3. Update again in case you have installed software that could get security updates from Pro repositories.
+
+No reboot is needed.
+
+
+Run the SSM script
+------------------
+
+The SSM script can be run either from the Fleet Manager or with the SSM Run Command. More information can be found in `this tutorial <https://ubuntu.com/tutorials/how-to-create-ssm-documents-and-use-them-to-install-packages-massively-with-run-command#1-overview>`
+
+You don’t need to restart the machines after applying the token. If it fails, check the SSM format of your document, see if there are any typos and try again. If this does not work check the output generated by SSM and see for any configuration issue (do your machines have access to update and upgrade packages? Pro repositories are available on different endpoints. Check the references at the end of the tutorial).
```

Comment:
```suggestion
You don’t need to restart the machines after applying the token. If the SSM script fails, check the SSM format of your document, see if there are any typos and try again. If this does not work check the output generated by SSM and look for any configuration issue (do your machines have access to update and upgrade packages? Pro repositories are available on different endpoints [[than what?]]. Check the references at the end of the tutorial).
```

What endpoints are Pro repositories available at? Are these static, and if so, can they be listed here?